public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -498,8 +498,10 @@ export const RoleMenuGroupOption = React.forwardRef<HTMLDivElement, RoleMenuGrou
 RoleMenuGroupOption.displayName = 'RoleMenuGroupOption';
 
 const getRoleGroup = (role: Role) => {
-  const parts = role.name.split(':');
-  return parts.length > 1 ? parts[1] : '';
+  if (role.group) {
+    return role.group;
+  }
+  return 'Other';
```

Comment:
Nit: Just a little more concise
```suggestion
  return role.group ?? 'Other';
```